pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -566,8 +569,8 @@ public LLRealtimeSegmentDataManager(RealtimeSegmentZKMetadata segmentZKMetadata,
 
 
     List<String> invertedIndexColumns = indexingConfig.getInvertedIndexColumns();
-    if (!indexingConfig.getSortedColumn().isEmpty()) {
-      invertedIndexColumns.add(indexingConfig.getSortedColumn().get(0));
+    if (_sortedColumn != null && !invertedIndexColumns.contains(_sortedColumn)) {
+      invertedIndexColumns.add(_sortedColumn);
```

Comment:
While in memory we want inverted index for the column in which queries are made. We don't keep the data sorted in memory, but sort it before creating the segment.